scripts/vendor_prost.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+readonly SCRIPTS_DIR="$(dirname "$0")"
+# shellcheck source=scripts/common
+source "${SCRIPTS_DIR}/common"
+
+# Updates our local copy of https://github.com/danburkert/prost and applies a small patch to add
+# support for the message_type field option.
+
+# Master branch as of 2020-06-05
+PROST_ZIP_URL="https://github.com/danburkert/prost/archive/80256277997975948d257faf3f35c2890bf12787.zip"
+PROST_ZIP_SHA256="02c086f2d1b7aad110745ac701f001618cd8343f882fc9ce034aa007c4c53773"
+PROST_ZIP_PREFIX="prost-80256277997975948d257faf3f35c2890bf12787"
+
+# Make sure we run in the third_party directory.
+cd "${SCRIPTS_DIR}/../third_party/prost" || exit
+
+# Download the prost zip only if it does not yet exist in the filesystem.
+if [ ! -f "$PROST_ZIP_PREFIX.zip" ]; then
+    rm -f "$PROST_ZIP_PREFIX.zip"
+    curl -L "$PROST_ZIP_URL" -o "$PROST_ZIP_PREFIX.zip"
+fi
+
+# Make sure the SHA256 checks out
+echo "$PROST_ZIP_SHA256 $PROST_ZIP_PREFIX.zip" | sha256sum -c -
+
+# Clean up from previous runs
+rm -rf "$PROST_ZIP_PREFIX" vendored/
+
+# Unzip and move into the vendored directory (stripping the prefix).
+unzip "$PROST_ZIP_PREFIX.zip"
+mv "$PROST_ZIP_PREFIX" vendored
+
+# We do not need the bundled protobuf because we already require users to have it installed.
+rm -r vendored/prost-build/third-party
+
+(
+  # Apply any local patches.
+  cd vendored/ || exit
+  for patchfile in ../*.patch; do
+    patch -p1 < "$patchfile"
+  done
+
+  # Check the local version builds, which also regenerates Cargo.lock
+  cargo build
```

Comment:
This is a good idea!

Could we maybe do something like `PROTOC=protoc cargo test` instead?

Reply:
Good idea, done.